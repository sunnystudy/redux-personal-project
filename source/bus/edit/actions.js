//Types
import { types } from "./types";

export const editActions = {
    //Sync
    startEditTodo: (todo) => {
        return {
            type:    types.START_EDIT_TODO,
            payload: todo,
        };
    },
    stopEditTodo: () => {
        return {
            type: types.STOP_EDIT_TODO,
        };
    },
};
