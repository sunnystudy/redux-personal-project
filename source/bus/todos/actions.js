//Types
import { types } from "./types";

export const todosActions = {
    //Sync
    fillTodos: (todos) => {
        return {
            type:    types.FILL_TODOS,
            payload: todos,
        };
    },
    createTodo: (todo) => {
        return {
            type:    types.CREATE_TODO,
            payload: todo,
        };
    },
    removeTodo: (todoId) => {
        return {
            type:    types.REMOVE_TODO,
            payload: todoId,
        };
    },
    updateTodo: (newTodoData) => {
        return {
            type:    types.UPDATE_TODO,
            payload: newTodoData,
        };
    },
    togglePriorityTodo: (newTodoData) => {
        return {
            type:    types.TOGGLE_PRIORITY_TODO,
            payload: newTodoData,
        };
    },
    toggleCompletedTodo: (newTodoData) => {
        return {
            type:    types.TOGGLE_COMPLETED_TODO,
            payload: newTodoData,
        };
    },
    allCompletedTodos: (todos) => {
        return {
            type:    types.ALL_COMPLETED_TODOS,
            payload: todos,
        };
    },
    //Async
    createTodoAsync: (todoMessage) => {
        return {
            type:    types.CREATE_TODO_ASYNC,
            payload: todoMessage,
        };
    },
    fetchTodosAsync: () => {
        return {
            type: types.FETCH_TODOS_ASYNC,
        };
    },
    removeTodoAsync: (todoId) => {
        return {
            type:    types.REMOVE_TODO_ASYNC,
            payload: todoId,
        };
    },
    togglePriorityTodoAsync: (newTodoData) => {
        return {
            type:    types.TOGGLE_PRIORITY_TODO_ASYNC,
            payload: newTodoData,
        };
    },
    toggleCompletedTodoAsync: (newTodoData) => {
        return {
            type:    types.TOGGLE_COMPLETED_TODO_ASYNC,
            payload: newTodoData,
        };
    },
    updateTodoAsync: (newTodoData) => {
        return {
            type:    types.UPDATE_TODO_ASYNC,
            payload: newTodoData,
        };
    },
    allCompletedTodosAsync: (todos) => {
        return {
            type:    types.ALL_COMPLETED_TODOS_ASYNC,
            payload: todos,
        };
    },
};
