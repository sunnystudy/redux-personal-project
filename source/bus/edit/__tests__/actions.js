//Actions
import { editActions } from "../actions";

describe('edit actions: ', () => {
    test('startEditTodo', () => {
        expect(editActions.startEditTodo(__.todo)).toMatchSnapshot();
    });
    test('stopEditTodo', () => {
        expect(editActions.stopEditTodo()).toMatchSnapshot();
    });
});
