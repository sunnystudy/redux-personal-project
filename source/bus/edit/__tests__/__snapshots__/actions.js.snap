// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit actions:  startEditTodo 1`] = `
Object {
  "payload": Object {
    "completed": false,
    "favorite": false,
    "id": "TEST_ID",
    "message": "TEST_MESSAGE",
  },
  "type": "START_EDIT_TODO",
}
`;

exports[`edit actions:  stopEditTodo 1`] = `
Object {
  "type": "STOP_EDIT_TODO",
}
`;
